src/controllers/wishlistController.js

```diff
@@ -0,0 +1,63 @@
+const mongoose = require('mongoose');
+const Wishlist = require('./wishlist');
+
+// Add an album to a wishlist
+async function addAlbumToWishlist(req, res) {
+  try {
+    const { wishlist_id, album_id } = req.params;
+
+    if (
+      !mongoose.Types.ObjectId.isValid(wishlist_id) ||
+      !mongoose.Types.ObjectId.isValid(album_id)
+    ) {
+      return res.status(400).json({ error: 'Invalid ID format' });
+    }
+
+    const wishlist = await Wishlist.findByIdAndUpdate(
+      wishlist_id,
+      { $addToSet: { albums: album_id } },
+      { new: true }
+    ).populate('albums');
+
+    if (!wishlist) {
+      return res.status(404).json({ error: 'Wishlist not found' });
+    }
+
+    res.json(wishlist);
+  } catch (error) {
+    res.status(500).json({ error: 'Could not update wishlist' });
+  }
```

Comment:
We don't need to wrap this in a `try/catch`, just let the error handler middlewre handle any unhandled errors:

```suggestion

    const { wishlist_id, album_id } = req.params;

    if (
      !mongoose.Types.ObjectId.isValid(wishlist_id) ||
      !mongoose.Types.ObjectId.isValid(album_id)
    ) {
      return res.status(400).json({ error: 'Invalid ID format' });
    }

    const wishlist = await Wishlist.findByIdAndUpdate(
      wishlist_id,
      { $addToSet: { albums: album_id } },
      { new: true }
    ).populate('albums');

    if (!wishlist) {
      return res.status(404).json({ error: 'Wishlist not found' });
    }

    res.json(wishlist);
```

Reply:
try/catch block removed , commit 76ce464606bb3213e7ad047e9371024ae8917774